examples/recreate_vm.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+"""
+Show how to recreate VM while maintaining VM permissions
+
+vm_name - VM to recreate
+"""
+
+import logging
+import time
+
+import ovirtsdk4 as sdk
+import ovirtsdk4.types as types
+import sys
+
+
+vm_name = 'vm'
+
+# Create the connection to the server:
+
+connection = sdk.Connection(
+    url='https://localhost.localdomain:8443/ovirt-engine/api',
+    username='admin@internal',
+    password='redhat123',
+    ca_file='Environments/ca.cer',
+    debug=True,
+    log=logging.getLogger(),
+)
+
+system_service = connection.system_service()
+
+vms_service = system_service.vms_service()
+
+vm = vms_service.list(search='name=%s' % vm_name)
+if not vm:
+    print('VM doesnt exist')
+    sys.exit(0)
+
+vm = vm[0]
+vm_template = connection.follow_link(vm.template)
+template_id = vm_template.id
+template_name = vm_template.name
+template_base_template = vm_template.version.base_template
+
+if vm.use_latest_template_version == True:
+    print('VM should not use latest template version')
+    sys.exit(0)
+
+if vm.status != types.VmStatus.DOWN:
+    print('VM status should be down')
+    sys.exit(0)
```

Comment:
that's one way to go, I'd say that stopping the vm might be a better one (under the assumption that when triggering this script you are no longer interested in that vm so you don't mind powering it off)